docs/LoaderImplementationInterface.md

```diff
@@ -0,0 +1,1081 @@
+Manager wherever practical.
+Typically, this is most important for ICDs, and the location clearly ties the
+ICD to a given device.
+The `SOFTWARE\Khronos\Vulkan\Drivers` location is the older method for locating
+ICDs, but is primary for non-ICD implementations.
+
+See the
+[Implementation Manifest File Format](#implementation-manifest-file-format)
+section for more details.
+
+
+### Implementation Discovery on Linux
+
+On Linux, the Vulkan loader will scan for implementation manifest files using
+environment variables or corresponding fallback values if the corresponding
+environment variable is not defined:
+
+<table style="width:100%">
+  <tr>
+    <th>Search Order</th>
+    <th>Directory/Environment Variable</th>
+    <th>Fallback</th>
+    <th>Additional Notes</th>
+  </tr>
+  <tr>
+    <td>1</td>
+    <td>$XDG_CONFIG_HOME</td>
+    <td>$HOME/config</td>
+    <td><b>This path is ignored when running an application with root or
+        super-user access</b>.<br/>
+        This is done to ensure that an application with root access does not
+        run layers that did not need root access to install.
+    </td>
+  </tr>
+  <tr>
+    <td>1</td>
+    <td>$XDG_CONFIG_DIRS</td>
+    <td>/etc/xdg</td>
+    <td></td>
+  </tr>
+  <tr>
+    <td>2</td>
+    <td>$SYSCONFDIR</td>
+    <td>/etc</td>
+    <td>Compile-time option set to possible location of implementations
+        installed from non-Linux-distribution-provided packages.
+    </td>
+  </tr>
+  <tr>
+    <td>3</td>
+    <td>$EXTRASYSCONFDIR</td>
+    <td>/etc</td>
+    <td>Compile-time option set to possible location of implementations
+        installed from non-Linux-distribution-provided packages.
+        Typically only set if $SYSCONFDIR is set to something other than /etc
+    </td>
+  </tr>
+  <tr>
+    <td>4</td>
+    <td>$XDG_DATA_HOME</td>
+    <td>$HOME/.local/share</td>
+    <td><b>This path is ignored when running an application with root or
+        super-user access</b>.<br/>
+        This is done to ensure that an application with root access does not
+        run layers that did not need root access to install.
+    </td>
+  </tr>
+  <tr>
+    <td>5</td>
+    <td>$XDG_DATA_DIRS</td>
+    <td>/usr/local/share/:/usr/share/</td>
+    <td></td>
+  </tr>
+</table>
+
+The directory lists are concatenated together using the standard platform path
+separator (:).
+The loader then selects each path, and applies the "/vulkan/icd.d" suffix onto
+each and looks in that specific folder for manifest files.
+
+The Vulkan loader will open each manifest file found to obtain the name or
+pathname of an implementation's shared library (".dylib") file.
+
+**NOTE** While the order of folders searched for manifest files is well
+defined, the order contents are read by the loader in each directory is
+[random due to the behavior of readdir](https://www.ibm.com/support/pages/order-directory-contents-returned-calls-readdir).
```

Comment:
I wonder whether it would be worthwhile to sort them lexicographically (`strcmp()` order) just to be able to say that there is *some* defined order?

Reply:
We have had complaints about attempting to do that, unfortunately.  Since that would mean some vendors would always appear above others.  We're working on other methods to provide a more balanced sorting of available devices that are returned from the implementations themselves.